### PR TITLE
feat: Added 'Publish from tag' pipeline for AVM

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added 'Publish from tag' pipeline for AVM
- Updated description of current publishing script

| Pipeline |
| - |
| [Test-Run in fork](https://github.com/AlexanderSehr/bicep-registry-modules/actions/runs/6995186301/job/19029748855) |

### CODE REVIEW

File: .github/workflows/avm.platform.publish.tag.yml
Comment (AlexanderSehr) [inline .github/workflows/avm.platform.publish.tag.yml:1]:
```suggestion
name: "avm.platform.publish-tag"
```

File: .github/workflows/avm.platform.publish.tag.yml
Comment (eriqua) [inline .github/workflows/avm.platform.publish.tag.yml:7]: Would it be helpful to add an example?

File: .github/workflows/avm.platform.publish.tag.yml
Comment (eriqua) [inline .github/workflows/avm.platform.publish.tag.yml:51]: Wondering about the location of this script. Should we keep it here or move it to the platform folder? Currently modifying the script would trigger all validation pipelines.

File: avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1
Comment (eriqua) [inline avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1:34]: As per what discussed some time ago, I'd suggest a repositoryroot input parameter with default value, and passing the repo root from the pipeline, leveraging the corrwsponding gh env var

File: .github/workflows/avm.platform.publish.tag.yml
Comment (AlexanderSehr) [reply .github/workflows/avm.platform.publish.tag.yml:7]: Maybe, but where? In the title of the input? Or as a comment in the code? 
There is an example in the utility script, and I guess we should add this to the AVM docs too. There is in general a User story open to describe the publishing process where this could be a part of.

Comment (eriqua) [reply .github/workflows/avm.platform.publish.tag.yml:7]: In the description

File: .github/workflows/avm.platform.publish.tag.yml
Comment (AlexanderSehr) [reply .github/workflows/avm.platform.publish.tag.yml:51]: Well it's a script that is run only by a pipeline for publishing. If we'd move it anywhere else I suspect that it would be rather confusing / hard to find afterwards.

However - we could also exclude it from the trigger / find a smart way to exclude specific scripts? Stupid example: if they'd have 'manual' in the name, they would no trigger runs? 

It's a general challenge. For this script in particular it may not matter much as I doubt it will change much (if at all). But the topic itself is something we could brainstorm.

File: .github/workflows/avm.platform.publish.tag.yml
Comment (AlexanderSehr) [reply .github/workflows/avm.platform.publish.tag.yml:7]: That will make it very long. But ok, your call

File: avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1]: Added to both publish scripts. While I fundamentally think it's a good idea - I'd like to discuss this if we want to do this on scale. On the upside, it allows you to provide whatever path you want in case you want to load dependent functions in isoluation from a custom location.
The downside I'd like to bring up is that we're introducing more and more dependencies on the exact folder structure (outside of e.g. the publishing). I.e., we're using now absolute paths instead of relative ones. If we were to rename e.g. the pipelines folder or something, we'd need to now update a lot more places than before. Not saying we shouldn't do this necessarily, but I'd at least like to question if we're doing us a favor.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1]: But they're not. We're just passing the repository root. If we'd want to make them less dependent, we'd need to pass each function's path.

File: .github/workflows/avm.platform.publish.tag.yml
Comment (eriqua) [reply .github/workflows/avm.platform.publish.tag.yml]: I'm still inclined to have all platform scripts in the platform folder and to not increase trigger complexity. So sure let's discuss offline.

Comment (AlexanderSehr) [reply .github/workflows/avm.platform.publish.tag.yml]: Fair enough - to me it's not really worth discussing further. I don't agree, but don't care enough either. I moved it now into a `platform` folder to unblock this PR.

File: avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1
Comment (eriqua) [reply avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1]: That is true. It would however result in less effort through search&replace vs wondering how many levels to go up and down in the hierarchy starting from the current script.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/publish/Publish-ModuleFromTagToPBR.ps1]: That is true.

**Comment by eriqua** [comment]
@AlexanderSehr looks good to me, only prettier between you and merge

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr looks good to me, only prettier between you and merge

My nemesis

**Review by eriqua — APPROVED** [review]
:ok:

**Pull request merged**
